components/logins/src/lib.rs

```diff
@@ -9,21 +9,41 @@
 mod error;
 mod login;
 
-mod db;
+pub mod db;
 pub mod schema;
 mod store;
 mod update_plan;
 mod util;
 
-mod ffi;
+uniffi_macros::include_scaffolding!("logins");
 
 // Mostly exposed for the sync manager.
+use crate::db::open_and_get_salt;
+use crate::db::open_and_migrate_to_plaintext_header;
 pub use crate::db::LoginDb;
 pub use crate::db::LoginStore;
+use crate::db::MigrationMetrics;
+use crate::db::MigrationPhaseMetrics;
 pub use crate::error::*;
 pub use crate::login::*;
 pub use crate::store::*;
 
-pub mod msg_types {
-    include!("mozilla.appservices.logins.protobuf.rs");
+#[derive(Clone, PartialEq)]
```

Comment:
nit: Consider moving this into `login.rs` and exposing the `login` module as public.

Reply:
I think we also want comments about why we have the 2 nearly identical structs. IIRC, this is really just due to the Guid/String mismatch - in which case we should see how we can abuse/enhance uniffi to do the right thing in these cases (probably as a followup, but maybe not, depending on how the timing of the other parts of this end up)

Also, I think `pub(crate)` would be sufficient here - I'm mildly surprised it logins.rs actually needs to use it (which is why any `pub` is needed) - if that's impossible to unwind then I agree with Lougenia about moving it - although there could also be a case made for moving it into its own source file if we can avoid logins.rs needing to use it at all - I kinds think login.rs needing to know about both structs is a but of a smell

Reply:
login.rs needs to know due to the From<> impl block to convert from Login and PasswordInfo (now LoginRecord). Also a big blocker in simply removing it completely is a uniffi limitation of not being able to [have both a dictionary and object at the same time](https://mozilla.github.io/uniffi-rs/udl/interfaces.html). IMO, I think a future release should get rid of the `Login` struct and just have the LoginRecord so that we can uniffi it easily -- or potentially change uniffi to allow for this paradigm somehow